mirrord/cli/src/execution.rs

```diff
@@ -304,14 +306,19 @@ impl MirrordExecution {
             }))
             .await
             .map_err(|_| {
-                CliError::InitialCommFailed("Failed to request remote environment variables.")
+                CliError::InitialCommFailed(
+                    "Failed to request remote environment variables.".to_string(),
+                )
             })?;
 
         match connection.receiver.recv().await {
             Some(DaemonMessage::GetEnvVarsResponse(Ok(remote_env))) => {
                 trace!("DaemonMessage::GetEnvVarsResponse {:#?}!", remote_env.len());
                 Ok(remote_env)
             }
+            Some(DaemonMessage::Close(msg)) => Err(CliError::InitialCommFailed(format!(
+                "Connection closed with message: {msg}"
```

Comment:
```suggestion
                "Connection closed with message: `{msg}`"
```

Just a nit to improve separation.